resdk/resolwe.py

```diff
@@ -241,6 +242,63 @@ def _process_file_field(self, path):
             'file_temp': file_temp,
         }
 
+    def _get_process(self, slug=None):
+        """Return process with given slug.
+
+        Raise error if process doesn't exist or more than one is returned
+        """
+        process = self.api.process.get(slug=slug, ordering='-version', limit=1)
+
+        if len(process) == 1:
+            process = process[0]
+        elif len(process) == 0:
+            raise ValueError("Could not get process for given slug.")
+        else:
+            raise ValueError("Unexpected behaviour at get process with slug {}".format(slug))
+
+        return process
+
+    def _process_inputs(self, inputs, process):
+        """Process input fields."""
+        inputs = copy.deepcopy(inputs)  # leave original intact
+
+        try:
+            for schema, fields in iterate_fields(inputs, process['input_schema']):
+                field_name = schema['name']
+                field_type = schema['type']
+                field_value = fields[field_name]
+
+                # Wrap `list:` fields into list if they are not already
+                if field_type.startswith('list:') and not isinstance(field_value, list):
+                    fields[field_name] = [field_value]
+                    field_value = fields[field_name]  # update value for the rest of the loop
+
+                # Dehydrate `data:*` fields
+                if field_type.startswith('data:'):
+                    fields[field_name] = get_data_id(field_value)
+
+                # Dehydrate `list:data:*` fields
+                elif field_type.startswith('list:data:'):
+                    fields[field_name] = []
+                    for data in field_value:
+                        fields[field_name].append(get_data_id(data))
+
+                # Upload files in `basic:file` fields
+                elif field_type == 'basic:file:':
+                    fields[field_name] = self._process_file_field(field_value)
+
+                # Upload files in list:basic:file` fields
+                elif field_type == 'list:basic:file:':
+                    fields[field_name] = []
+                    for obj in field_value:
+                        fields[field_name].append(self._process_file_field(obj))
+
+        except KeyError as key_error:
+            raise KeyError("Field '{}' not in process '{}' input schema.".format(key_error.args[0],
```

Comment:
`ValidationError` is Django specific, it is not defined in Python
https://docs.python.org/3/library/exceptions.html